### ISSUE TITLE
MATLAB.jl: Error loading MATLAB Module in OSX

### ISSUE THREAD

**Issue by GravityAssisted** [JuliaLang/MATLAB.jl#27]
Hi am new to Julia and have been trying to use MATLAB.jl. It installed just fine using the `Pkg.add("MATLAB")` command, but when I type `using MATLAB` , I get the following error:

```
ERROR: LoadError: LoadError: MethodError: `min` has no method matching min(::Array{Union(UTF8String,ASCIIString),1})
Closest candidates are:
  min(::Any, ::Any)
  min(::Any, ::Any, ::Any)
  min(::Any, ::Any, ::Any, ::Any...)
 in get_paths at /Users/arora/.julia/v0.4/MATLAB/src/mxbase.jl:20
 in include at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include_from_node1 at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include_from_node1 at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in reload_path at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in _require at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in require at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
while loading /Users/arora/.julia/v0.4/MATLAB/src/mxbase.jl, in expression starting on line 66
while loading /Users/arora/.julia/v0.4/MATLAB/src/MATLAB.jl, in expression starting on line 34
```

When I try running commands like ` restart_default_msession() ` I get errors lile `ERROR: UndefVarError: restart_default_msession not defined`

Do you know what could be going wrong here and how can this be fixed ? I am on Julia 0.4.  thanks a lot

**Comment by GravityAssisted** [comment]
I fixed this issue using the git cloned version instead of the `Pkg.add()` , but now I have a new error showing up:

```
ERROR: LoadError: LoadError: UndefVarError: RTLD_GLOBAL not defined
 in load_libmx at /Users/arora/.julia/v0.4/MATLAB/src/mxbase.jl:76
 in include at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include_from_node1 at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in include_from_node1 at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in reload_path at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in _require at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
 in require at /Users/arora/CODES/Julia_source/julia/usr/lib/julia/sys.dylib
while loading /Users/arora/.julia/v0.4/MATLAB/src/mxbase.jl, in expression starting on line 84
while loading /Users/arora/.julia/v0.4/MATLAB/src/MATLAB.jl, in expression starting on line 34
```

I am trying to find a solution to this so any help is very welcomed :-) . thanks

**Comment by GravityAssisted** [comment]
So.. looks like I was able to track this problem down to `RTLD_GLOBAL | RTLD_LAZY` parameters. I think something must have changed from 0.3 to 0.4. The following call fails (with the above error):

```libmx = dlopen(matlab_library("libmx"), RTLD_GLOBAL | RTLD_LAZY)```

and I had to remove and `RTLD_GLOBAL | RTLD_LAZY` replace this command by

```libmx = dlopen(matlab_library("libmx"))```

...similarly for other `dlopen` function calls in `mxbase.jl`. After I did that, the module seems to load fine. Any clarifications on this are welcomed ?

thanks.
